examples/internal/proto/examplepb/stream.pb.gw.go

```diff
@@ -306,12 +310,20 @@ func RegisterStreamServiceHandlerClient(ctx context.Context, mux *runtime.ServeM
 			runtime.HTTPError(ctx, mux, outboundMarshaler, w, req, err)
 			return
 		}
-		resp, md, err := request_StreamService_BulkEcho_0(annotatedContext, inboundMarshaler, client, req, pathParams)
+
+		resp, md, reqErrChan, err := request_StreamService_BulkEcho_0(annotatedContext, inboundMarshaler, client, req, pathParams)
 		annotatedContext = runtime.NewServerMetadataContext(annotatedContext, md)
 		if err != nil {
 			runtime.HTTPError(annotatedContext, mux, outboundMarshaler, w, req, err)
 			return
 		}
+		go func() {
+			for err := range reqErrChan {
+				if err != nil && err != io.EOF {
+					runtime.HTTPError(annotatedContext, mux, outboundMarshaler, w, req, err)
```

Comment:
Is it safe to call this function with the shared `w`? Is there some way we can avoid doing this concurrently (a defer? or something).

Reply:
so testing this implementation with my server works fine but I agree that might be more optimal solution.

I'm not sure about the concern around sharing `w`, the timing of this invalid argument error and the timing of a stream of valid responses could happen concurrently.  I don't see any better indicators for when it's appropriate to send the error the client.

But one thing I noticed is that `runtime.HTTPError` modifies the header, which means the header could be modified twice.  I'm thinking about writing the error directly using `w.Write()`. My thinking is that if the error occurs in the middle of the stream, it makes sense that the original response header stays at 200.

I have been trying to modify the [integration test](https://github.com/grpc-ecosystem/grpc-gateway/blob/b18dc950f6c8952da46f8a88689f0e8e190a2e05/examples/internal/integration/integration_test.go#L1341) to test it out but I found that the current test set up waits for the entire request body to be written before actually making the http request. I would expect the response collection to start without waiting for the first goroutine of requests writing to complete. Unfortunately I haven't figured out why that isn't the case yet so it's hard to test the scenario of error occurring in the middle of a stream.

Reply:
It's fine if we can't easily test it. I don't think setting the header will matter once we've started writing to the body, since we can't write the header again. Given that, this is probably fine :). Thanks for testing it manually.

Reply:
Hey Johan, thanks for approving this, but I still find that using `ErrorHandler` might not be ideal. PTAL at https://github.com/grpc-ecosystem/grpc-gateway/pull/4864. Let's move the discussion over as well